test/functional/rpc_scanblockfilters.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test the scanblocks rpc call."""
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal, assert_raises_rpc_error
+
+class scanblocksTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 2
+        self.extra_args = [["-blockfilterindex=1"],[]]
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def run_test(self):
+        # send 1.0, mempool only
+        addr_1 = self.nodes[0].getnewaddress()
+        self.nodes[0].sendtoaddress(addr_1, 1.0)
+
+        # send 1.0, mempool only
+        self.nodes[0].sendtoaddress("mkS4HXoTYWRTescLGaUTGbtTTYX5EjJyEE", 1.0) # childkey 5 of tpubD6NzVbkrYhZ4WaWSyoBvQwbpLkojyoTZPRsgXELWz3Popb3qkjcJyJUGLnL4qHHoQvao8ESaAstxYSnhyswJ76uZPStJRJCTKvosUCJZL5B
+
+        # mine a block and assure that the mined blockhash is in the filterresult
+        blockhash = self.nodes[0].generate(1)[0]
```

Comment:
In 71b7cdb460e2d0179aa87bdd0d30a82a821f6d05 "Add scanblock functional test"

Since the indexes are built asynchronously, it is possible that after generating the block, the block filter index has not updated to include the new block, so scanblocks might not find anything. Instead, we need to wait for it to become synced:

```suggestion
        blockhash = self.nodes[0].generate(1)[0]
        self.wait_until(lambda: all(i["synced"] for i in self.nodes[0].getindexinfo().values()))
```

Reply:
Fixed in #23549